src/App.vue

```diff
@@ -1,26 +1,57 @@
 <template>
-  <img alt="Vue logo" src="./assets/logo.png">
-  <HelloWorld msg="Welcome to Your Vue.js App"/>
+<input :value="text" ref="myInput" />
+<button @click="focusOnInput">Focus</button>
+<button class="text-white rounded bg-pink-500 px-6 py-4 font-sans font-semibold shadow-lg hover:bg-pink-400 outline-none active:outline-none mt-5" @click="toggleModal = !toggleModal">Open Modal</button>
+
+<MyFirstModal modalTitle="Sign Up for the GiveAway" modalContent="Grab Your ninja swag for half price" theme="sale" :show-modal="toggleModal" @updateShowModal = "toggleModal = $event">
```

Comment:
same here @updateShowModal = "toggleModal = $event" 
do not change the value in the template. @guptasakshi282001